gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -1947,3 +1929,91 @@ def test_unsupported_parameter_rest_asyncio():
+# and should replace it once we cover all the test cases in it here.
+#}
+{% macro rest_transport_specific_tests(service, transport, is_async) %}
+{% if 'rest' in transport %}
+{### Test syntax configuration begins ###}
+{% set await_prefix = "await " if is_async else "" %}
+{% set async_prefix = "async " if is_async else "" %}
+{% set async_decorator = "@pytest.mark.asyncio " if is_async else "" %}
+{% set transport_name = get_transport_name(transport, is_async) %}
+{### Test syntax configuration ends ###}
+{### Test generation for all methods begins ###}
+{% for method in service.methods.values() %}
+{% set method_name = method.name|snake_case %}
+{### test cases section begin ###}
+{% if is_rest_unsupported_method(method, is_async) == 'True' or not method.http_options %}
+{{ rest_method_not_implemented_error(service, method, method_name, transport_name, async_prefix, await_prefix, async_decorator, is_async) }}
+{% endif %}{# is_rest_unsupported_method(method, is_async) == 'False' and method.http_options #}
+{### test cases section ends ###}
+{% endfor %}
+{### Test generation for all methods ends ###}
+{{ rest_method_initialize_client_test(service, transport_name, is_async) }}
+{% endif %}{# if 'rest' in transport #}
+{% endmacro %}
+
+{# rest_method_not_implemented_error generates tests for methods
+ # which are not supported for rest transport.
+#}
+{% macro rest_method_not_implemented_error(service, method, method_name, transport_name, async_prefix, await_prefix, async_decorator, is_async) %}
+{% if not is_async %}{# TODO(b/362949446): Remove this guard once a not implemented __call__ method is added to async rest for all methods. #}
+{{async_decorator}}
+{{async_prefix}}def test_{{ method_name }}_{{transport_name}}_error():
+    {% if transport_name == 'rest_asyncio' %}
+    if not HAS_GOOGLE_AUTH_AIO:
+        {# TODO(https://github.com/googleapis/google-auth-library-python/pull/1577): Update the version of google-auth once the linked PR is merged. #}
+        pytest.skip("google-auth > 2.x.x is required for async rest transport.")
+    {% endif %}
+
+    client = {{ get_client(service, is_async) }}(
+        credentials={{get_credentials(is_async)}},
+        transport="{{transport_name}}"
+    )
+
+    # Since a `google.api.http` annotation is required for using a rest transport
+    # method, this should error.
+    with pytest.raises(NotImplementedError) as not_implemented_error:
+        {{await_prefix}}client.{{ method_name }}({})
+    assert (
+        "Method {{ method.name }} is not available over REST transport"
+        in str(not_implemented_error.value)
+    )
+
+{% endif %}{# if is_async #}
+{% endmacro %}
+
+{# rest_method_initialize_client_test adds coverage for rest transport clients.
+ # Note: Unlike before, this test case is needed because we aren't unconditionally 
+ # generating the not implemented coverage test for every client.
+#}
+{% macro rest_method_initialize_client_test(service, transport_name, is_async) %}
```

Comment:
This test could be made generic where it runs for both `grpc` and `rest`

Reply:
addressed.